examples/mri/non_cartesian_reconstruction.py

```diff
@@ -2,133 +2,96 @@
+from mri.parallel_mri.extract_sensitivity_maps import \
+    gridded_inverse_fourier_transform_nd
+import pysap
+from pysap.data import get_sample_data
 
 # Third party import
+from modopt.math.metrics import ssim
 import numpy as np
-import scipy.fftpack as pfft
-
 
 # Loading input data
-image = get_sample_data("mri-slice-nifti")
-image.data += np.random.randn(*image.shape) * 20.
-mask = get_sample_data("mri-mask")
+image = get_sample_data('2d-mri')
+
+# Obtain MRI non-cartesian mask
+radial_mask = get_sample_data("mri-radial-samples")
+kspace_loc = radial_mask.data
+mask = pysap.Image(data=convert_locations_to_mask(kspace_loc, image.shape))
+
+# View Input
 image.show()
 mask.show()
 
-
 #############################################################################
 # Generate the kspace
 # -------------------
 #
-# From the 2D brain slice and the acquistion mask, we generate the acquisition
-# measurments, the observed kspace.
-# We then reconstruct the zero order solution.
+# From the 2D brain slice and the acquisition mask, we retrospectively
+# undersample the k-space using a radial acquisition mask
+# We then reconstruct the zero order solution as a baseline
 
 # Get the locations of the kspace samples and the associated observations
-kspace_loc = convert_mask_to_locations(mask.data)
 fourier_op = NFFT(samples=kspace_loc, shape=image.shape)
 kspace_obs = fourier_op.op(image.data)
 
-# Zero order solution
-image_rec0 = pysap.Image(data=fourier_op.adj_op(kspace_obs),
-                         metadata=image.metadata)
+# Gridded solution
+grid_space = np.linspace(-0.5, 0.5, num=image.shape[0])
+grid2D = np.meshgrid(grid_space, grid_space)
+grid_soln = gridded_inverse_fourier_transform_nd(kspace_loc, kspace_obs,
+                                                 tuple(grid2D), 'linear')
+image_rec0 = pysap.Image(data=grid_soln)
 image_rec0.show()
-
+base_ssim = ssim(image_rec0, image)
+print('The Base SSIM is : ' + str(base_ssim))
 
 #############################################################################
 # FISTA optimization
 # ------------------
 #
 # We now want to refine the zero order solution using a FISTA optimization.
-# Here no cost function is set, and the optimization will reach the
-# maximum number of iterations. Fill free to play with this parameter.
+# The cost function is set to Proximity Cost + Gradient Cost
 
 # Generate operators
 gradient_op, linear_op, prox_op, cost_op = generate_operators(
     data=kspace_obs,
-    wavelet_name="BsplineWaveletTransformATrousAlgorithm",
+    wavelet_name=24,
     samples=kspace_loc,
+    mu=6 * 1e-7,
     nb_scales=4,
     non_cartesian=True,
     uniform_data_shape=image.shape,
     gradient_space="synthesis")
 
 # Start the FISTA reconstruction
-max_iter = 20
-x_final, transform, costs, metrics = sparse_rec_fista(
-    gradient_op,
-    linear_op,
-    prox_op,
-    None,
-    mu=1e-9,
+max_iter = 200
+x_final, costs, metrics = sparse_rec_fista(
```

Comment:
Note that the current implementation of this function is passes `verbose=True` to `costObj` (see [here](https://github.com/CEA-COSMIC/ModOpt/blob/9f50074a8f0e888edb97b74f1c4d277874b5e8ef/modopt/opt/cost.py#L79)), which prints the cost at every iteration. I think it's best for the examples to simply use the progress bar.

Reply:
Well, we would still want to see the cost function decrease, although I agree with you. I think it is a matter of audience to whom these examples are for. 
In my opnion these examples are to demonstrate the features and not the performance gains, so it makes sense to show how cost function is also tracked over iterations.
Again, @zaccharieramzi and @philouc to comment more on this

Reply:
Well we don't necessarily want it printed out at every iteration. That's the metrics variable is here for if we want afterwards to plot the cost function at every iteration. `verbose` is more for debugging purposes.

Reply:
Well that would be a new PR for pysap-mri. I feel this is important, I will get one.
I will add verbose levels, and enable the verbosity only if the verbose to main function is `>=10`

Reply:
https://github.com/CEA-COSMIC/pysap-mri/pull/47 PR in place, resolving here